File: app/src/main/java/com/cere/csplayer/adapter/RecyclerViewListener.java
package com.cere.csplayer.adapter;

import android.view.GestureDetector;
import android.view.MotionEvent;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.core.view.GestureDetectorCompat;
import androidx.recyclerview.widget.RecyclerView;

import com.simplecityapps.recyclerview_fastscroll.views.FastScrollRecyclerView;

/**
 * Created by CheRevir on 2019/10/24
 */
public class RecyclerViewListener implements FastScrollRecyclerView.OnItemTouchListener {
    private GestureDetectorCompat mGestureDetectorCompat;

    public RecyclerViewListener(final RecyclerView recyclerView, final OnItemListener onItemListener) {
        mGestureDetectorCompat = new GestureDetectorCompat(recyclerView.getContext(), new GestureDetector.SimpleOnGestureListener() {
            @Override
            public boolean onSingleTapUp(MotionEvent e) {
                View childView = recyclerView.findChildViewUnder(e.getX(), e.getY());
                if (childView != null) {
                    onItemListener.onItemClick(recyclerView.getChildLayoutPosition(childView));
                }
                return true;
            }

            @Override
            public void onLongPress(MotionEvent e) {
                super.onLongPress(e);
                View childView = recyclerView.findChildViewUnder(e.getX(), e.getY());
                if (childView != null) {
                    onItemListener.onItemLongClick(recyclerView.getChildLayoutPosition(childView));
                }
            }
        });
    }

    @Override
    public boolean onInterceptTouchEvent(@NonNull RecyclerView rv, @NonNull MotionEvent e) {
        mGestureDetectorCompat.onTouchEvent(e);
        return false;
    }

    @Override
    public void onTouchEvent(@NonNull RecyclerView rv, @NonNull MotionEvent e) {

    }

    @Override
    public void onRequestDisallowInterceptTouchEvent(boolean disallowIntercept) {

    }

    public interface OnItemListener {
        void onItemClick(int position);

        void onItemLongClick(int position);
    }

}

File: app/src/main/java/com/cere/csplayer/adapter/RecyclerViewListener.kt
package com.cere.csplayer.adapter

import android.view.GestureDetector
import android.view.MotionEvent
import androidx.core.view.GestureDetectorCompat
import androidx.recyclerview.widget.RecyclerView

/**
 * Created by CheRevir on 2020/9/16
 */
class RecyclerViewListener(
    private val recyclerView: RecyclerView,
    private val listener: OnRecyclerItemListener
) :
    GestureDetector.SimpleOnGestureListener(), RecyclerView.OnItemTouchListener {
    private var detector: GestureDetectorCompat = GestureDetectorCompat(recyclerView.context, this)

    override fun onSingleTapUp(e: MotionEvent?): Boolean {
        return e?.let {
            val view = recyclerView.findChildViewUnder(e.x, e.y)
            view?.let { v ->
                listener.onItemClick(recyclerView.getChildAdapterPosition(v))
                true
            } ?: false
        } ?: false
    }

    override fun onLongPress(e: MotionEvent?) {
        super.onLongPress(e)
        e?.let {
            val view = recyclerView.findChildViewUnder(e.x, e.y)
            view?.let { v ->
                listener.onItemLongClick(recyclerView.getChildAdapterPosition(v))
            }
        }
    }

    override fun onInterceptTouchEvent(rv: RecyclerView, e: MotionEvent): Boolean {
        detector.onTouchEvent(e)
        return false
    }

    override fun onTouchEvent(rv: RecyclerView, e: MotionEvent) {
    }

    override fun onRequestDisallowInterceptTouchEvent(disallowIntercept: Boolean) {
    }

    interface OnRecyclerItemListener {
        fun onItemClick(position: Int)
        fun onItemLongClick(position: Int)
    }
}